distribution_metadata.c

```diff
@@ -419,28 +370,30 @@ PartitionType(Oid distributedTableId)
 bool
 IsDistributedTable(Oid tableId)
 {
+	Oid metadataNamespaceOid = get_namespace_oid("pgs_distribution_metadata", false);
+	Oid tableNamespaceOid = get_rel_namespace(tableId);
 	bool isDistributedTable = false;
-	RangeVar *heapRangeVar = NULL;
-	Relation heapRelation = NULL;
-	HeapScanDesc scanDesc = NULL;
-	const int scanKeyCount = 1;
-	ScanKeyData scanKey[scanKeyCount];
-	HeapTuple heapTuple = NULL;
-
-	heapRangeVar = makeRangeVar(METADATA_SCHEMA_NAME, PARTITION_TABLE_NAME, -1);
-	heapRelation = relation_openrv(heapRangeVar, AccessShareLock);
+	Oid argTypes[] = { OIDOID };
+	Datum argValues[] = { ObjectIdGetDatum(tableId) };
+	const int argCount = sizeof(argValues) / sizeof(argValues[0]);
+	int spiStatus = 0;
 
-	ScanKeyInit(&scanKey[0], ATTR_NUM_PARTITION_RELATION_ID, InvalidStrategy,
-				F_OIDEQ, ObjectIdGetDatum(tableId));
+	if (tableNamespaceOid == metadataNamespaceOid)
```

Comment:
>> Fundamentally, this check should only be needed for the partition table, is that correct?
> Correct. Since its rows are the top-level objects in the graphs of distribution metadata for tables, putting the short-circuit here is sufficient.

If that's the case, should we make the check more precise by only checking if relation name = 'pgs_distribution_metadata.partition'? That makes the comment and its context clearer?